cmd/cli/plugin/cluster/upgrade.go

```diff
@@ -181,6 +181,12 @@ func getValidTKRVersionForUpgradeGivenFullTKRName(clusterName, namespace string,
 		return tkrForUpgrade.Spec.Version, nil
 	}
 
+	// // If existing TKR on the cluster and the one requested are same
+	// // continue the upgrade using the given TKR
+	if tkrName == tkrForUpgrade.GetName() {
+		return tkrForUpgrade.Spec.Version, nil
+	}
+
```

Comment:
This LGTM. The only problem (nitpick, really) I have with this is this function is already too long, and is just getting longer, definitely more complex and more difficult to understand.